js.go

```diff
@@ -798,14 +798,28 @@ func AckExplicit() SubOpt {
 	})
 }
 
-func MaxWaiting(n int) {
+func AckWait(d time.Duration) SubOpt {
```

Comment:
Wonder if this could be confusing with `nats.MaxWait`, when publishing maybe someone could try to do:

```
js.Publish("foo", nats.AckWait(2*time.Second))
```

right now to get that behavior would need to do:

```
js.Publish("foo", nats.MaxWait(2*time.Second))
```

Reply:
Ok. Removed `configurePublish` from `MaxWait` and made `AckWait` a `SubOpt` and a `PubOpt` aaaaand also updated the publish tests to use `AckWait` instead of `MaxWait`.